shared/src/balancer/pool_fetching.rs

```diff
@@ -0,0 +1,80 @@
+use anyhow::Result;
+use model::TokenPair;
+use std::collections::HashSet;
+
+use crate::balancer::event_handler::{BalancerPoolStore, WeightedPool};
+use crate::pool_fetching::{handle_contract_error, Block, MAX_BATCH_SIZE};
+use crate::Web3;
+use contracts::BalancerV2Vault;
+use ethcontract::batch::CallBatch;
+use ethcontract::errors::MethodError;
+use ethcontract::{BlockId, Bytes, H160, U256};
+use itertools::Itertools;
+
+pub struct FetchedWeightedPool {
+    pub pool: WeightedPool,
+    /// getPoolTokens returns (Tokens, Balances, LastBlockUpdated)
+    pub reserves: Result<(Vec<H160>, Vec<U256>, U256), MethodError>,
```

Comment:
This should probably not be of type Result (but instead the values directly, and we just don't have a pool if the reserves fail).

I think this struct should be named WeightedPool (the other one is really just static metadata about the pool). Also maybe we can expose the important information in a flat data structure (right now tokens are e.g. exposed twice). Maybe @fedgiac can already give some insights on how this struct would be most conveniently used for the math he is working on?

Reply:
RIght, this was really only meant to be a temp struct that is used before the results are handled. This struct has since been renamed `PreFetchedWeighted` pool and made non-public. The recent update has its reserves field as just `Vec<U256>`. 

On your other point, I will see how I can rename the other weighted pool to accommodate the rename here and flatten the struct.

Reply:
I have included a doc comment on this struct:

```
/// only an internal temporary struct used during pool fetching to handle any errors when fetching.
```

I hope this helps clear it up. Will refactor and flatted the publicly exposed struct for @fedgiac.

Reply:
In terms of usability, a HashMap would probably be best, since starting from in and out tokens we need the in and out balances for only these tokens.

Reply:
So something like 
```rust
reserves: `HashMap<Token, PoolBalance>`
```